File: cpp/source/event/input/user/OpenChannels.cpp
#include "event/input/user/OpenChannels.hpp"

#include <sstream>
#include <iterator>

using namespace fm;
using namespace fm::event::input::user;

using namespace com::fleetmgr::interfaces;

OpenChannels::OpenChannels(std::shared_ptr<std::vector<std::shared_ptr<ChannelRequest>>> _channels) :
    UserEvent(OPEN_CHANNELS),
    channels(_channels)
{
}

const std::vector<std::shared_ptr<ChannelRequest>>& OpenChannels::getChannels() const
{
    return *channels;
}

std::string OpenChannels::toString() const
{
    std::ostringstream oss;
    oss << "[";
    for (std::shared_ptr<ChannelRequest> c : *channels)
    {
        oss << c->id() << ",";
    }
    oss << "]";
    return "OPEN_CHANNELS: channels: " + oss.str();
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/input/user/OpenChannels.java
package com.fleetmgr.sdk.client.event.input.user;

import com.fleetmgr.interfaces.ChannelRequest;

import java.util.List;


/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class OpenChannels extends UserEvent {

    private List<ChannelRequest> channels;

    public OpenChannels(List<ChannelRequest> channels) {
        super(Type.OPEN_CHANNELS);
        this.channels = channels;
    }

    public List<ChannelRequest> getChannels() {
        return channels;
    }

    @Override
    public String toString() {
        return "OpenChannels{" +
                "channels=" + channels +
                '}';
    }
}
